File: fyn-flutter-app/lib/features/message/data/models/create_conversation_request.dart
import 'conversation_type.dart';

class CreateConversationRequest {
  final Set<String> participantIds;
  final String? title;
  final ConversationType type;

  CreateConversationRequest({
    required this.participantIds,
    this.title,
    this.type = ConversationType.direct,
  });

  Map<String, dynamic> toJson() {
    return {
      'participantIds': participantIds.toList(),
      if (title != null) 'title': title,
      'type': type.serverValue,
    };
  }
}

File: fyn-monolithic/src/main/java/com/fyn_monolithic/dto/request/message/CreateConversationRequest.java
package com.fyn_monolithic.dto.request.message;

import com.fyn_monolithic.model.message.ConversationType;
import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

import java.util.Set;

@Data
public class CreateConversationRequest {

    @NotEmpty
    private Set<String> participantIds;

    private String title;

    private ConversationType type = ConversationType.DIRECT;
}
